test/socket.test.js

```diff
@@ -0,0 +1,171 @@
+const Websocket = require('ws');
+const Errors = require('../lib/errors');
+const Socket = require('../lib/socket').ConstellationSocket;
+const port = process.env.SERVER_PORT || 1339;
+
+describe('socket', () => {
+    let server;
+    let socket;
+    const url = `ws://127.0.0.1:${port}/`;
+
+    beforeEach(ready => {
+        server = new Websocket.Server({ port }, ready);
+    });
+
+    afterEach(done => {
+        if (socket) {
+            socket.close();
+            socket = null;
+        }
+
+        server.close(done);
+    });
+
+    describe('connecting', () => {
+        it('connects with no auth', done => {
+            socket = new Socket({ url }).connect();
+            server.on('connection', ws => {
+                expect(ws.upgradeReq.url).to.equal('/');
+                expect(ws.upgradeReq.headers.authorization).to.be.undefined;
+                done();
+            });
+        });
+
+        it('connects with JWT auth', done => {
+            socket = new Socket({ url, jwt: 'asdf!' }).connect();
+            server.on('connection', ws => {
+                expect(ws.upgradeReq.url).to.equal('/?jwt=asdf!');
+                expect(ws.upgradeReq.headers.authorization).to.be.undefined;
+                done();
+            });
+        });
+
+        it('connects with an OAuth token', done => {
+            socket = new Socket({ url, authToken: 'asdf!' }).connect();
+            server.on('connection', ws => {
+                expect(ws.upgradeReq.url).to.equal('/');
+                expect(ws.upgradeReq.headers.authorization).to.equal('Bearer asdf!');
+                done();
+            });
+        });
+
+        it('throws an error on ambiguous auth', () => {
+            expect(() => new Socket({ url, authToken: 'asdf!', jwt: 'wat?' })).to.throw();
```

Comment:
To clarify, the fact that something throws doesn't mean it throws the right thing.